23_multi-assay_analyses.Rmd

```diff
@@ -158,6 +160,120 @@ ggplot(correlation_table, aes(x = Var1, y = Var2, fill = cor)) +
   labs(x = "Taxa", y = "Metabolites")
 ```
 
+## Multi-Omics Factor Analysis
+
+Multi-Omics Factor Analysis [@Argelaguet2018] (MOFA) is
+an unsupervised method for integrating multi-omic data sets in a downstream analysis.
+It could be seen as a generalization of principal component analysis. Yet, with the ability to infer a latent (low-dimensional) representation, 
+shared among the mutliple (-omics) data sets in hand.
+
+We use the R [MOFA2](https://biofam.github.io/MOFA2/index.html) package
+for the analysis, and [install](https://biofam.github.io/MOFA2/installation.html) the corresponding dependencies.
+
+**Note:** Comment out the line for installing `miniconda` if it has been installed earlier.
+Similarly for `mofapy2` if it was so.
+
+```{r, message=FALSE, warning=FALSE}
+if(!require(MOFA2)){
+    BiocManager::install("MOFA2")
+}
+
+# For inter-operability between Python and R, and setting Python dependencies,
+# reticulate package is needed
+if(!require(reticulate)){
+    install.packages("reticulate")
+}
+
+reticulate::install_miniconda(force = TRUE)
+reticulate::use_miniconda()
+reticulate::py_install(packages = c("mofapy2"), pip = TRUE)
+```
+
+The `mae` object could be used straight to create the MOFA model:
+
+```{r, message=FALSE, warning=FALSE}
+library(MOFA2)
+# For simplicity, classify all high-fat diets as high-fat, and all the low-fat 
+# diets as low-fat diets
+colData(mae)$Diet <- ifelse(colData(mae)$Diet == "High-fat" | 
+                              colData(mae)$Diet == "High-fat + XOS", 
+                            "High-fat", "Low-fat")
+
+# Removing duplicates at the microbiome data
+# which are also in form e.g. "Ambiguous" and "uncultured" taxa
+mae[[1]] <- mae[[1]][!duplicated(rownames(assay(mae[[1]]))), ]
+
+# Transforming microbiome data with clr
+mae[[1]] <- transformCounts(mae[[1]], abund_values = "counts", method = "clr",
+                            pseudocount = 1)
```

Comment:
Does the result change if we use rclr? That is potentially better than pseudocount+clr but we do not have much experience with it yet.

Reply:
I have not tried it yet, I will do it with rclr.

Reply:
it did change results considerably (first one with clr, second with rclr):
![image](https://user-images.githubusercontent.com/86964562/149949775-81bc5e26-6fe0-4902-95b5-957c06933506.png)
![image](https://user-images.githubusercontent.com/86964562/149949847-9fdf66a0-9882-46ad-b0df-d0c2d3aab112.png)

Reply:
Interesting. I am wondering what MOFA assumes in general about the input data. Would count data be even more appropriate? Does the MOFA have any recommendations?

Reply:
The recommendations from MOFA are gaussian likelihoods (the two other options are poisson or bernoulli)

Reply:
are all three data sets approximately Gaussian for the features?

Reply:
I unfortunately just by now checked for normality, the two first assays looks approximately normal but not the third:
![image](https://user-images.githubusercontent.com/86964562/150149155-1c98bb49-988a-431f-935f-4dd904ceaa91.png)

Reply:
When I do the features z-transform of the biomarker data I get this:
![image](https://user-images.githubusercontent.com/86964562/150150281-723633a8-9b54-4206-97f8-63ffa04f96f8.png)
Shall I use that?

Reply:
Yes. Also mention in the text that this is done because the model assumes normality; and just mention those other options that were available.